rclcpp_components/cmake/rclcpp_components_register_node.cmake

```diff
@@ -0,0 +1,69 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# usage: rclcpp_components_register_node(
+#              <target> PLUGIN <component> EXECUTABLE <node>)
+#
+# Register an rclcpp component with the ament
+# resource index and create an executable.
+#
+# :param target: the shared library target
+# :type target: string
+# :param PLUGIN: the plugin name
+# :type PLUGIN: string
+# :type EXECUTABLE: the node's executable name
+# :type EXECUTABLE: string
+#
+macro(rclcpp_components_register_node target)
+  cmake_parse_arguments(
+    ARGS
+    ""
+    "PLUGIN;EXECUTABLE"
+    ""
+    ${ARGN})
+  set(component ${ARGS_PLUGIN})
+  set(node ${ARGS_EXECUTABLE})
+  _rclcpp_components_register_package_hook()
+  set(_path "lib")
+  if(WIN32)
+    set(_path "bin")
+    set(library_name ${target}.dll)
+  elseif(APPLE)
+    set(library_name lib${target}.dylib)
+  else()
+    set(library_name lib${target}.so)
+  endif()
```

Comment:
Instead of trying to guess the library name can't it be retrieved from the library target instead? I am worried about cases not handled here, e.g. debug suffix etc.

Reply:
@dirk-thomas is this somewhat like what you had in mind?

Reply:
:+1: